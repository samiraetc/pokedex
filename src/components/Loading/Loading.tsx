import image from '../../assets/gifs/loading.gif'

export const COMPONENT_ID = 'pokedex-loading';

const Loading = () => {
    return (
        <div id={COMPONENT_ID} className={COMPONENT_ID}>
            <img src={image} alt="pokebola icon" className='pokedex-loading__icon' />
        </div>
    )
}

export default Loading;