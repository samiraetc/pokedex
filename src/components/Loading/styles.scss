.pokedex-loading {
    position: absolute;
    top: 55%;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%);

    &__icon {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        width: 300px;
    }
}

