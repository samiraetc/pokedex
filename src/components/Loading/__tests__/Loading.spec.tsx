import { render, configure, fireEvent } from '@testing-library/react'
import Loading, { COMPONENT_ID } from '../Loading';

configure({
    testIdAttribute: 'id',
  });
  
describe('[LOADING]', () => {
    test('should render Loading', () => {
        const { asFragment, getByTestId } = render(<Loading />);

        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});