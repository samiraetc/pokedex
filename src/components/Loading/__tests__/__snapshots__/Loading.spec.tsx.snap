// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LOADING] should render Loading 1`] = `
<DocumentFragment>
  <div
    class="pokedex-loading"
    id="pokedex-loading"
  >
    <img
      alt="pokebola icon"
      class="pokedex-loading__icon"
      src="default"
    />
  </div>
</DocumentFragment>
`;
